EIPS/eip-2537.md

```diff
@@ -269,21 +269,35 @@ Assuming a constant `30 MGas/second`, the following prices are suggested.
 
 #### G2 multiplication
 
-`45000` gas
+`22500` gas
 
 #### G1/G2 MSM
 
 MSMs are expected to be performed by Pippenger's algorithm (we can also say that it **must** be performed by Pippenger's algorithm to have a speedup that results in a discount over naive implementation by multiplying each pair separately and adding the results). For this case there was a table prepared for discount in case of `k <= 128` points in the MSM with a discount cap `max_discount` for `k > 128`.
 
 To avoid non-integer arithmetic, the call cost is calculated as `(k * multiplication_cost * discount) / multiplier` where `multiplier = 1000`, `k` is a number of (scalar, point) pairs for the call, `multiplication_cost` is a corresponding single multiplication call cost for G1/G2.
 
-Discounts table as a vector of pairs `[k, discount]`:
+G1 and G2 are priced separately, each having their own discount table and `max_discount`.
+
+##### G1 discounts
+
+Discounts table for G1 MSM as a vector of pairs `[k, discount]`:
+
+```
+[[1, 1800], [2, 888], [3, 955], [4, 802], [5, 743], [6, 821], [7, 750], [8, 680], [9, 657], [10, 635], [11, 612], [12, 591], [13, 664], [14, 637], [15, 611], [16, 585], [17, 578], [18, 571], [19, 564], [20, 557], [21, 628], [22, 620], [23, 612], [24, 604], [25, 596], [26, 588], [27, 578], [28, 570], [29, 562], [30, 693], [31, 683], [32, 673], [33, 670], [34, 665], [35, 663], [36, 658], [37, 655], [38, 650], [39, 648], [40, 643], [41, 640], [42, 635], [43, 633], [44, 628], [45, 625], [46, 620], [47, 618], [48, 613], [49, 610], [50, 605], [51, 603], [52, 598], [53, 595], [54, 590], [55, 588], [56, 583], [57, 580], [58, 578], [59, 573], [60, 570], [61, 565], [62, 563], [63, 558], [64, 555], [65, 553], [66, 550], [67, 548], [68, 548], [69, 545], [70, 543], [71, 540], [72, 540], [73, 538], [74, 535], [75, 533], [76, 533], [77, 530], [78, 528], [79, 528], [80, 525], [81, 523], [82, 520], [83, 520], [84, 518], [85, 515], [86, 513], [87, 513], [88, 510], [89, 508], [90, 505], [91, 505], [92, 503], [93, 500], [94, 498], [95, 498], [96, 495], [97, 493], [98, 490], [99, 490], [100, 488], [101, 485], [102, 483], [103, 483], [104, 480], [105, 478], [106, 478], [107, 475], [108, 473], [109, 470], [110, 470], [111, 468], [112, 465], [113, 463], [114, 463], [115, 460], [116, 458], [117, 455], [118, 455], [119, 453], [120, 450], [121, 448], [122, 448], [123, 445], [124, 443], [125, 440], [126, 440], [127, 438], [128, 435]]
```

Comment:
For `k=1` we should put 1000 (the same cost as multiplication). Otherwise, we incentivize contracts to put a condition to redirect inputs from MSM to MUL in case they have single point. 
In other words, if you have single point to multiply, you should not use any MSM algorithm. Just use regular multiplication in your MSM precompile implementation.

Reply:
These values are not monotonic, so you have to fix this.

Reply:
For me the terminal value `k=128` is `520`.

Reply:
if we put 1000 instead of 1800 I think G1MSM will be too slow for GNARK

Reply:
> These values are not monotonic, so you have to fix this.

missed that.  they should definitely be monotonic, regardless of the performance profile for any particular k,